src/main/java/core/basesyntax/strategy/SupplyHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyHandler implements OperationHendler {
+    @Override
+    public void operateTransaction(FruitTransaction transaction) {
+        int oldQuantity = Storage.fruitStorage.getOrDefault(transaction.getFruit(), 0);
+        Storage.fruitStorage.put(transaction.getFruit(), oldQuantity + transaction.getQuantity());
```

Comment:
Extract `fruitStorage.put(transaction.getFruit(), oldQuantity + transaction.getQuantity()` to the new method `Storage.put(String fruit, Integer quantity)` in class Storage. For better compliance to Single Responsibility principe. 
Call this new methods (instead of direct operation with map) in every Handler class where it needed.